lib/sendf.c

```diff
@@ -254,6 +255,43 @@ ssize_t Curl_send_plain(struct connectdata *conn, int num,
                         const void *mem, size_t len, CURLcode *code)
 {
   curl_socket_t sockfd = conn->sock[num];
+#ifdef _WIN32
+  /* WinSock will destroy unread received data if send() is
+     called after detection of disconnect.
+     To avoid lossage of received data, recv() must be
+     performed before every send() if any incoming data is
+     available. */
+  if((conn->handler->protocol&PROTO_FAMILY_HTTP) != 0 &&
+      conn->recv[num] == Curl_recv_plain) {
+    const int readymask = Curl_socket_check(sockfd, CURL_SOCKET_BAD,
+                                            CURL_SOCKET_BAD, 0);
```

Comment:
It should be rare situation.
However you're right, `Curl_socket_check()` is not expensive call, but it produce a system call, so better to avoid it if possible.